src/main/java/lesson_jdbc_01/CustomerDaoImplGeneric.java

```diff
@@ -0,0 +1,52 @@
+package lesson_jdbc_01;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+public class CustomerDaoImplGeneric {
+
+	public boolean insertCustomer(Customer customer) throws SQLException {
+		return new TemplateAbstract<Customer>() {
+			@Override
+			public PreparedStatement returnPreparedStatement(Customer customer, Connection conn) throws SQLException {
+				PreparedStatement stmt;
+				conn = ConnectToDB.getConnection();
```

Comment:
Is it necessary to get connection in each method? You can pass connection as parameter.

Reply:
Yes, better to template